setup.py

```diff
@@ -42,6 +42,7 @@
         "web3==4.3.0",
         # required for rlp>=1.0.0
         "eth-account>=0.2.1,<1",
+        "cachetools>=2.1.0"
```

Comment:
@zixuanzh almost there! I think this should have an upper bound to not break things accidentially when versions change. Maybe even lock it down to "cachetools==2.1.*" which will only allow higher patch versions.

Reply:
@cburgdorf updated, thanks!

Reply:
@cburgdorf consistency wise, I think this should have been `cachetools>=2.1.0,<3` (assuming we don't have a specific reason to stay within patch versions).  

We've taken a loose policy with dependencies, **assuming** proper semver which I think is an *ok* assumption at this stage, meaning that we blindly accept minor/patch version bumps, but restrict on the major version.

Reply:
I'm sorry I messed up the consistency there. I sent a PR to fix it.

https://github.com/ethereum/py-evm/pull/999

>We've taken a loose policy with dependencies, assuming proper semver which I think is an ok assumption at this stage, meaning that we blindly accept minor/patch version bumps, but restrict on the major version.

Personally, I've been biten to often by dependencies that were doing breaking changes in non-major version increases that I became more conservative in this regard. My preferred solution for applications (not libraries) is actually to pin down exact versions via lock files.

Reply:
> My preferred solution for applications (not libraries) is actually to pin down exact versions via lock files.

I won't be surprised if this is where we end up.  I think I'd like to try an intermediate step towards pinning against `minor` ranges and see how things shake out.